apps/astarte_data_updater_plant/test/astarte_data_updater_plant/triggers_handler_test.exs

```diff
@@ -334,6 +339,54 @@ defmodule Astarte.DataUpdaterPlant.TriggersHandlerTest do
       assert Map.get(headers_map, static_header_key) == static_header_value
     end
 
+    test "incoming_introspection AMQPTarget handling with legacy event" do
+      Config.put_generate_legacy_incoming_introspection_events(true)
+      simple_trigger_id = :uuid.get_v4()
+      parent_trigger_id = :uuid.get_v4()
+      static_header_key = "important_metadata_incoming_introspection"
+      static_header_value = "test_meta_incoming_introspection"
+      static_headers = [{static_header_key, static_header_value}]
+      timestamp = get_timestamp()
+
+      target = %AMQPTriggerTarget{
+        simple_trigger_id: simple_trigger_id,
+        parent_trigger_id: parent_trigger_id,
+        static_headers: static_headers,
+        routing_key: @routing_key
+      }
+
+      TriggersHandler.incoming_introspection(
+        target,
+        @realm,
+        @device_id,
+        @introspection,
+        timestamp,
+        nil
+      )
+
+      assert_receive {:event, payload, meta}
+
+      assert %SimpleEvent{
+               device_id: @device_id,
+               parent_trigger_id: ^parent_trigger_id,
+               simple_trigger_id: ^simple_trigger_id,
+               realm: @realm,
+               timestamp: ^timestamp,
+               event: {:incoming_introspection_event, incoming_introspection_event}
+             } = SimpleEvent.decode(payload)
+
+      assert %IncomingIntrospectionEvent{introspection: @introspection} =
+               incoming_introspection_event
+
+      headers_map = amqp_headers_to_map(meta.headers)
+
+      assert Map.get(headers_map, "x_astarte_realm") == @realm
+      assert Map.get(headers_map, "x_astarte_device_id") == @device_id
+      assert Map.get(headers_map, "x_astarte_event_type") == "incoming_introspection_event"
+      assert Map.get(headers_map, static_header_key) == static_header_value
+      Config.put_generate_legacy_incoming_introspection_events(false)
```

Comment:
Put this at the beginning in an `on_exit` callback:
```elixir
Config.put_generate_legacy_incoming_introspection_events(true)
on_exit(fn -> Config.put_generate_legacy_incoming_introspection_events(false))
```

Otherwise if the test terminates early it won't restore this, possibly leading to more tests failing (and making it hard to track the actual only cause of the failure)